sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java

```diff
@@ -288,8 +289,8 @@ public DirectorySetMetadataInfo setMetadata(Map<String, String> metadata) {
      *                   If the request does not specify maxresults or specifies a value greater than 5,000, the server will return up to 5,000 items.
      * @return {@link FileRef File info} in this directory with prefix and max number of return results.
      */
-    public Iterable<FileRef> listFilesAndDirectories(String prefix, Integer maxResults) {
-        return directoryAsyncClient.listFilesAndDirectories(prefix, maxResults).toIterable();
+    public PagedIterable<FileRef> listFilesAndDirectories(String prefix, Integer maxResults) {
```

Comment:
We need timeouts in the sync cases. You can look at similar operations in the blob service for an example on how to do it. Make sure you check all the sync clients for this.

Reply:
Created issue #5189